examples/plot_vector_testing.py

```diff
@@ -0,0 +1,158 @@
+    label = ctype
+    lines.plot(color=color, ax=ax, label=label)
+
+size = 0
+
+for ctype, points in point_gdf.groupby('attr2'):
+    color = point_symb[ctype]
+    label = ctype
+    size += 100
+    points.plot(color=color, ax=ax, label=label, markersize=size)
+
+# Add a legend
+ax.legend(title="Legend", loc=(1.1, .1));
+
+# If you were running this in a notebook, the commented out  line below would
+# store the matplotlib object. However, in this example, you can just grab the
+# axes object directly.
+
+# plot_1_hold = nb.convert_axes(plt, which_axes="current")
+
+################################################################################
+# Create A VectorTester Object
+# --------------------------------
+# Once you've created your plot, you can create a MatPlotCheck VectorTester object
+# that can be used to test elements in the plot.
+
+vector_test = VectorTester(ax)
+
+###############################################################################
+#
+# .. note::
+#   Each geometry type must be tested seperately in VectorTester. So, if your
+#   plot has multiple geometry types, such as lines, polygons, and points,
+#   make sure to check each geometry type seperately.
+
+###############################################################################
+#
+# .. note::
+#   If a test fails, matplotcheck will return an error. If the test passes,
+#   no message is returned.
+
+################################################################################
+# Testing Point Attribute Values and Geometry
+# -------------------------------------------
+# You can check that both the position of the points on the plot and the associated
+# point attribute values are
+# accurate using assert_points(), assert_points_grouped_by_type() and
+# assert_collection_sorted_by_markersize. If the plot uses point markers that are
+# sized by attribute value, you can check that the
+# size of each marker correctly relates to an attribute value.
+
+# Check point geometry location (x, y location)
+vector_test.assert_points(point_gdf)
+
+# Check points are grouped plotted by type
+vector_test.assert_points_grouped_by_type(point_gdf, "attr2")
+
+# Check points size is relative to a numeric attribute value
+vector_test.assert_collection_sorted_by_markersize(point_gdf, "attr2")
+
+################################################################################
+# Test Line Attribute Values and Coordinate Information (x, y)
+# ------------------------------------------------------------
+# You can also test the position and plot values of line
+# geometries.
+
+# Check line geometry
+vector_test.assert_lines(multi_line_gdf)
+
+# Check lines plotted by type
+vector_test.assert_lines_grouped_by_type(multi_line_gdf, "attr")
+
+################################################################################
+# Testing Polygon Geometry
+# ------------------------
+# Currently, MatPlotCheck is unable to check that polygons were plotted by type.
```

Comment:
@nkorinek just checking. We discussed you adding this or creating an issue, is that correct?

Reply:
Yup! Issue is here: https://github.com/earthlab/matplotcheck/issues/225

Reply:
perfect. thanks~